src/pubsub-adapter.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+// Pubsub adapter to keep API with handlers while not removed.
+module.exports = (PubsubRouter, libp2p, options) => {
+  class Pubsub extends PubsubRouter {
+    /**
+     * Subscribes to a given topic.
+     * @override
+     * @param {string} topic
+     * @param {function(msg: InMessage)} [handler]
+     * @returns {void}
+     */
+    subscribe (topic, handler) {
+      super.subscribe(topic)
+
+      // Bind provided handler
+      handler && this.on(topic, handler)
```

Comment:
It shouldn't matter, but we should bind the listener first. This is also consistent with the examples for the new api.

```suggestion      
      // Bind provided handler
      handler && this.on(topic, handler)
      super.subscribe(topic)
```